hibernate-core/src/main/java/org/hibernate/dialect/MariaDBDialect.java

```diff
@@ -34,35 +34,34 @@
  */
 public class MariaDBDialect extends MySQLDialect {
 
-	private final DatabaseVersion version;
+	private final DatabaseVersion mariaVersion;
 
 	public MariaDBDialect() {
 		this( DatabaseVersion.make( 5 ) );
 	}
 
 	public MariaDBDialect(DialectResolutionInfo info) {
-		this( info.makeCopy(), getCharacterSetBytesPerCharacter( info.unwrap( DatabaseMetaData.class ) ) );
+		this( info.makeCopy(), info );
 		registerKeywords( info );
 	}
 
 	public MariaDBDialect(DatabaseVersion version) {
-		// Let's be conservative and assume people use a 4 byte character set
-		this( version, 4 );
+		this(version, null);
 	}
 
-	public MariaDBDialect(DatabaseVersion version, int characterSetBytesPerCharacter) {
+	protected MariaDBDialect(DatabaseVersion mariaVersion, DialectResolutionInfo info) {
 		super(
-				version.isBefore( 5, 3 )
+				mariaVersion.isBefore( 5, 3 )
 						? DatabaseVersion.make( 5 )
 						: DatabaseVersion.make( 5, 7 ),
-				characterSetBytesPerCharacter
+				info
 		);
-		this.version = version;
+		this.mariaVersion = mariaVersion;
 	}
 
 	@Override
 	public DatabaseVersion getVersion() {
-		return version;
+		return mariaVersion;
```

Comment:
I believe this could become a problem because that will return null when the `columnType` methods are called from within the `super` constructor call.

Reply:
That's right, but the `mariaVersion` is never (or at least _should_ never be used for anything. It should always be `getMySQLVersion()` that is used. (A bit confusing, I know.)

Reply:
Let's continue this here: https://github.com/hibernate/hibernate-orm/pull/4451#discussion_r767914434